qiskit/visualization/state_visualization.py

```diff
@@ -522,7 +522,12 @@ def plot_state_paulivec(
 ):
     """Plot the paulivec representation of a quantum state.
 
-    Plot a bargraph of the mixed state rho over the pauli matrices
+    Plot a bargraph of the density matrix of a quantum state using as a basis all
+    possible tensor products of Pauli operators and identities, that is,
```

Comment:
```suggestion
    Plot the expectation values of the quantum state against measurement bases of all
    possible tensor products of Pauli operators and identities, that is,
```
Is this more accurate description?

Reply:
We are mostly talking about the same thing expressed in different ways. What you state is accurate except for a factor $2^n$.

Let $\rho =\sum_{i} \sum_{j} \alpha_{ij} P_i \otimes P_j$ be a 2-qubit density matrix, where $i,j\in$ {0,1,2,3}, $P_0=I$, $P_1=X$, $P_2=Y$, $P_3=Z$. This can always be done because the Pauli matrices and identity form a basis of complex $2\times 2$ matrices.

The expected value of the operator $P_k\otimes P_l$ is given by $\langle P_k\otimes P_l \rangle=\mbox{Tr}(\rho\cdot A) = \sum_{i} \sum_{j} \alpha_{ij} \mbox{Tr}(P_kP_i \otimes P_lP_j) = \sum_{i} \sum_{j} \alpha_{ij} \mbox{Tr}(P_iP_k)\mbox{Tr}(P_jP_l)$

Using that Pauli matrices $(X, Y, Z)$ are traceless while identity has trace 2, the product of 2 different Paulis is another Pauli (multiplied by a constant $\pm i$) and that $P_iP_i = I$ $\forall i$, the only term that is not cancelled in the expected value is $\alpha_{kl} \mbox{Tr}(P_kP_k)\mbox{Tr}(P_lP_l) = \alpha_{kl} \mbox{Tr}(I)\mbox{Tr}(I) = 2^2\alpha_{kl}$

Reply:
Thanks for the detailed explanations. Now I understand the meaning of the plot! It is basically plotting the coefficient of the quantum state expressed as a series of pauli operators like the [SparsePauliOperator](https://qiskit.org/documentation/stubs/qiskit.quantum_info.SparsePauliOp.html?highlight=sparsepauliop#qiskit.quantum_info.SparsePauliOp).

In that case we should change the y axis which states "expectation values" to "coefficient".

Reply:
I am also wondering whether we include identity as one of the Pauli matrices / operators. Strictly speaking it is not part of them, but when we say expressing a state in Pauli matrices / operators, we almost always include identity as well. So I am not sure we need to say that explicitly. 

@Cryoris would you like to take a look at the improved explanation?

Reply:
I think there's an inconsistency with the y-label which says "Expectation value": if you look at the first example in the docstring what's being plotted is what Guillermo said, but what the y-label suggests is what Junye wrote 😅 I'd suggest to leave Guillermo's explanation and update the y-axis label, this way we leave the plot values as they were and do the more subtle correction.

Reply:
Changed y-axis labels to "coefficients" instead of "expectation values" in [db9c27f](https://github.com/Qiskit/qiskit-terra/pull/8420/commits/db9c27f9164d29eed05a7135d16ecea590ec41b0)